pgmpy/inference/Sampling.py

```diff
@@ -132,17 +142,30 @@ def rejection_sample(self, evidence=None, size=1):
         """
         if evidence is None:
             return self.forward_sample(size)
-        sampled = DataFrame(columns=self.topological_order)
+        types = [(var_name, 'int') for var_name in self.topological_order]
+        sampled = np.zeros(0, dtype=types).view(np.recarray)
         prob = 1
-        while len(sampled) < size:
-            _size = int(((size - len(sampled)) / prob) * 1.5)
+        i = 0
+        while i < size:
+            _size = int(((size - i) / prob) * 1.5)
             _sampled = self.forward_sample(_size)
+
             for evid in evidence:
-                _sampled = _sampled[_sampled.ix[:, evid[0]] == evid[1]]
+                _sampled = _sampled[_sampled[evid[0]] == evid[1]]
+
             prob = max(len(_sampled) / _size, 0.01)
-            sampled = sampled.append(_sampled)
-        sampled.reset_index(inplace=True, drop=True)
-        return sampled[:size]
+
+            if HAS_PANDAS is True:
```

Comment:
@yashu-seth I find this behaviour to be ambiguous. Lets say I have got two envs one contains pandas and other just specific to pgmpy. Both envs has pgmpy installed. Now if I create a script in first env which has pandas and didn't specify the parameter for return type, later I switched the env and tried to run my script it will raise an error that will be difficult for user understand, like indexing of this type not supported etc. But with the case I'm suggesting it will raise error pandas not found. And also there is no extra step in specifying a parameter.